### PR TITLE
build: `make clean` now only deletes dist directories

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlike to be merged.
* [x] Optional. I've added My organization is added to the README.
* [x] I've signed the CLA and required builds are green. 

See #2016 

How to have a terminal prompt that displays your current context.

```
source /usr/local/share/kube-ps1.sh
export PS1='[\u@\h \W $(kube_ps1)]\$ '
```

How to add k3d to your list of contexts:

```
export KUBECONFIG="$(k3d get-kubeconfig --name='k3s-default'):$HOME/.kube/config"
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,11 +44,14 @@ VERSION = ${GIT_TAG}
 override LDFLAGS += -X ${PACKAGE}.gitTag=${GIT_TAG}
 endif
 
+KUBECTX          := $(shell kubectl config current-context)
+ALLOWED_KUBECTXS := docker-desktop|docker-for-desktop|minikube
```
Comment (alexec) [inline Makefile]: must check this is a correct/complete list

File: Makefile [hunk 2]
```diff
@@ -218,8 +221,16 @@ dist/postgres.yaml: test/e2e/manifests/postgres.yaml
 	# Create Postgres e2e manifests
 	cat test/e2e/manifests/postgres.yaml | sed 's/:latest/:$(IMAGE_TAG)/' > dist/postgres.yaml
 
+local:
```
Comment (alexec) [inline Makefile]: * [ ] Test by setting kubectx to production system and run `make local`, expect exit code 1.

File: Makefile [hunk 3]
```diff
@@ -45,6 +45,9 @@ VERSION = ${GIT_TAG}
 override LDFLAGS += -X ${PACKAGE}.gitTag=${GIT_TAG}
 endif
 
+KUBECTX          := $(shell kubectl config current-context)
+ALLOWED_KUBECTXS := docker-desktop|docker-for-desktop|minikube|k3s-default
```
Comment (simster7) [inline Makefile]: This doesn't support named `minikube` instances. For example, if a start `minikube` with `minikube start -p test` then the output of `kubectl config current-context` would be `test`.

Comment (alexec) [reply Makefile]: do you use a named instance?

Comment (alexec) [reply Makefile]: that said, I think we just need to decide on the correct test here, kubectx might not be it.

Comment (alexec) [reply Makefile]: what about this

```
kubectl config view --minify | grep 'server:' | grep -o localhost
```

**Comment by alexec** [comment]
> ```
> export KUBECONFIG="$(k3d get-kubeconfig --name='k3s-default'):$HOME/.kube/config"
> ```

m

File: Makefile [hunk 4]
```diff
@@ -239,8 +239,17 @@ dist/postgres.yaml: test/e2e/manifests/postgres.yaml
 	# Create Postgres e2e manifests
 	cat test/e2e/manifests/postgres.yaml | sed 's/:latest/:$(IMAGE_TAG)/' | sed 's/pns/$(E2E_EXECUTOR)/' > dist/postgres.yaml
 
+.PHONY: local
+local:
+ifneq ($(CI),true)
+ifeq ($(findstring localhost,`kubectl config view --minify | grep 'server:' | grep -o localhost`),)
```
Comment (simster7) [inline Makefile]: Looks like this doesn't work either :(. On my local:

```
$ kubectl config view --minify | grep 'server:'
    server: https://192.168.99.110:8443
```

**Pull request merged**